### PR TITLE
test(vhost-user-blk): add a helper for drive resizing

### PR DESCRIPTION
**Pull request by kalyazin**
## Changes

This includes a `crosvm` vhost-user-blk backend in the devctr and adds a test helper to trigger vhost-user-blk device resizes.

## Reason

We would like to test that Firecracker handles config changes in vhost-user-blk drives properly.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- ~[ ] If a specific issue led to this PR, this PR closes the issue.~
- [x] The description of changes is clear and encompassing.
- ~[ ] Any required documentation changes (code and docs) are included in this PR.~
- ~[ ] API changes follow the [Runbook for Firecracker API changes][2].~
- ~[ ] User-facing changes are mentioned in `CHANGELOG.md`.~
- [x] All added/changed functionality is tested.
- ~[ ] New `TODO`s link to an issue.~
- [x] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [x] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

File: tests/framework/utils_vhost_user_backend.py [hunk 1]
```diff
@@ -33,3 +50,12 @@ def spawn_vhost_user_backend(vm, host_mem_path, socket_path, readonly=False):
...
     return proc
+
+
+def resize(new_size, backend="crosvm"):
+    """Resize vhost-user-blk drive and send config change notification"""
+
+    if backend != "crosvm":
+        assert False, f"backend `{backend}` does not support resizing"
+
+    utils.run_cmd(f"crosvm disk resize 0 {new_size} {CROSVM_CTR_SOCKET}")
```
Comment (pb8o) [inline tests/framework/utils_vhost_user_backend.py:61]: Would it make sense to have a few classes to keep it organized?

```python
class VhostUserBackend(ABC):
    def __init__(self, socket: Path):
        self.socket = socket

    @abstractmethod
    def spawn(self):
        pass

    def resize(self):
        raise NotImplementedError()

    @staticmethod
    def with_backend(backend, *args, **kwargs):
        if backend == "qemu":
            return VhostUserBackendQEmu(*args, **kwargs)
        ...

class VhostUserBackendQEmu(VhostUserBackend):
    def spawn(self):
        pass

class VhostUserBackendCrosvm(VhostUserBackend):
    def resize(self):
        pass

> VhostUserBackend.with_backend("qemu", socket="/my.sock")
> <__main__.VhostUserBackendQEmu object at 0x7f70773300d0>
```

Comment (kalyazin) [reply tests/framework/utils_vhost_user_backend.py:61]: do you think that would be worth it, provided that we only have 2 methods (spawn and resize), and resize will only be supported by one of the backends?

File: tests/framework/utils_drive.py [hunk 2]
```diff
@@ -38,18 +38,40 @@ def partuuid_and_disk_path(rootfs_ubuntu_22, disk_path):
 
 
 VHOST_USER_SOCKET = "/vub.socket"
+CROSVM_CTR_SOCKET = "/crosvm_ctr.socket"
 
 
-def spawn_vhost_user_backend(vm, host_mem_path, socket_path, readonly=False):
+def spawn_vhost_user_backend(
+    vm, host_mem_path, socket_path, readonly=False, backend="qemu"
```
Comment (roypat) [inline tests/framework/utils_drive.py:45]: nit: Could `backend` be an enum instead of a string?

File: tests/framework/utils_drive.py [hunk 3]
```diff
@@ -63,3 +85,17 @@ def spawn_vhost_user_backend(vm, host_mem_path, socket_path, readonly=False):
...
+    Resize vhost-user-blk drive and send config change notification.
+
+    New size is in MB.
+    """
+
+    if backend != "crosvm":
+        assert False, f"backend `{backend}` does not support resizing"
+    assert os.path.exists(CROSVM_CTR_SOCKET), "crosvm backend must be spawned first"
+
+    utils.run_cmd(f"crosvm disk resize 0 {new_size * 1024 * 1024} {CROSVM_CTR_SOCKET}")
```
Comment (roypat) [inline tests/framework/utils_drive.py:101]: This having a `backend` parameter only for it to check that the parameter has a specific value feels a bit weird. Isnt the "path exists" check enough here?

File: tests/framework/utils_drive.py [hunk 4]
```diff
@@ -38,18 +38,40 @@ def partuuid_and_disk_path(rootfs_ubuntu_22, disk_path):
 
 
 VHOST_USER_SOCKET = "/vub.socket"
+CROSVM_CTR_SOCKET = "/crosvm_ctr.socket"
```
Comment (kalyazin) [inline tests/framework/utils_drive.py:41]: A note to myself: this needs to be unique to work when parallelised.

File: tests/framework/utils_drive.py (hunk 2 above)
Comment (kalyazin) [reply tests/framework/utils_drive.py]: done

File: tests/framework/utils_drive.py (hunk 3 above)
Comment (kalyazin) [reply tests/framework/utils_drive.py]: done

File: tests/framework/utils_drive.py (hunk 4 above)
Comment (kalyazin) [reply tests/framework/utils_drive.py]: done

**Review by roypat — APPROVED** [review]

**Review by ShadowCurse — APPROVED** [review]

**Review by pb8o — APPROVED** [review]

**Pull request merged**